### ISSUE TITLE
Devstack doesn't use gunicorn

### ISSUE THREAD

**Issue by timmc-edx** [edx/devstack#92]
Unlike 2U's deployed environments, devstack uses Django's `runserver` command instead of gunicorn.

A/C:

- [ ] If feasible, switch devstack to use gunicorn
- [ ] If not, add docs on how to use gunicorn when desired, and why it's not the default

Notes:

- edx-platform's `lms/docker_lms_gunicorn.py` has a hardcoded `workers = 17` which may not be appropriate for devstack. We may just be able to change it, depending on what actually uses this file these days.
- Can we still get autoreloading with gunicorn?

**Comment by robrap** [comment]
@timmc-edx: I know you've been experimenting with gunicorn in devstack. What's the status of that? Are there temporary docs for that that would be used as part of this ticket? Is there anything you could add as a comment here? Thank you.

**Comment by timmc-edx** [comment]
For the new codejail service, I'm fairly sure I could just switch over the Dockerfile to just use gunicorn for both the `dev` and `prod` targets. It should just work. However, I think the main limiting factor is that `runserver` gives us auto-reloading when code changes on disk, and gunicorn doesn't, out of the box. So that's probably the main thing to look into.

**Comment by robrap** [comment]
Thanks @timmc-edx. That is partially answering my question about what is different about with and without gunicorn, but I wasn't actually asking about codejail. I was asking because there is another issue in DD in the LMS that may require local testing with gunicorn, and I wanted to link to a document for that setup.

**Comment by timmc-edx** [comment]
Yeah, but that's the service I happen to know about. :-) I thought you were asking about why we couldn't switch over _all_ services in devstack.

As for local testing with gunicorn in the LMS, I recommend https://github.com/edx/devstack/commit/e93d674774dc04d7c4f58e1dc0357a55caadd56f which is on the `timmc/lms-gunicorn` branch, itself based on the local-devstack branch. I don't have a proper document for it.

**Comment by robrap** [comment]
Right. This ticket is for docs, so it is ok that there aren’t any. Your link to the commit is great. If you have any other rough notes/thoughts that would be useful for someone else to pick this up, just add an additional comment on this ticket. Either way, this is great. Thank you.

**Comment by jristau1984** [comment]
https://2u-internal.atlassian.net/browse/BOMS-156
